### PR TITLE
Refactoring of ConvertExistenceCheckTransformer

### PR DESCRIPTION
**Pull request by b-scholz**
The classes of ConvertExistenceCheckTransfomer were removed and replaced by succinct visit/lambda constructs. 
This is related to issue #961.

### CODE REVIEW

File: src/RamTransforms.h [hunk 1]
```diff
@@ -129,28 +129,36 @@ class MakeIndexTransformer : public RamTransformer {
...
+ *
+ * will be rewritten to
+ *
+ *  QUERY
+ *   ...
+ *    IF (10,20) NOT IN A
+ *      ...
+ *
+ */
+
```
Comment (mmcgr) [inline src/RamTransforms.h]: Remove the extra line
```suggestion
```

File: src/RamTransforms.cpp [hunk 2]
```diff
@@ -268,165 +268,62 @@ bool MakeIndexTransformer::makeIndex(RamProgram& program) {
...
-        }
-    };
-
-    // level all RAM inserts
-    RamQueryCapturer insertUpdate(this);
-    program.getMain()->apply(insertUpdate);
-
-    return insertUpdate.getModified();
+        };
+        ((RamNode*)&query)->apply(makeLambdaRamMapper(scanRewriter));
```
[the remark this answers is not in the document]
Comment (b-scholz) [reply src/RamTransforms.cpp]: I was not sure whether visitDepthFirst accepts non-const casts. We are lucky - it works.

File: src/RamTransforms.cpp [hunk 3]
```diff
@@ -268,165 +268,62 @@ bool MakeIndexTransformer::makeIndex(RamProgram& program) {
...
-                } else if (const auto* intrinsicOp = dynamic_cast<const RamIntrinsicOperator*>(val)) {
-                    for (const RamExpression* arg : intrinsicOp->getArguments()) {
-                        queue.push_back(arg);
-                    }
-                } else if (const auto* userDefinedOp = dynamic_cast<const RamUserDefinedOperator*>(val)) {
-                    for (const RamExpression* arg : userDefinedOp->getArguments()) {
-                        queue.push_back(arg);
-                    }
-                }
+/** rewrite IndexScan to a filter/existence check if the IndexScan's tuple
```
[the remark this answers is not in the document]
Comment (b-scholz) [reply src/RamTransforms.cpp]: I have rewritten the comments in doxygen format. We should enforce this in future.

File: src/RamTransforms.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (b-scholz) [reply src/RamTransforms.cpp]: No - it does not because smart pointers can only be used as const pointers etc. etc.

**Comment by b-scholz** [comment]
Martin - can you fix the FreeBSD issue. There is no space left on the device.

**Comment by mmcgr** [comment]
Coredumps were taking up all the space. cleaned up now

**Pull request merged**